File: 409-longest-palindrome/longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map <char,int> mp;
        for(int i = 0;i < s.size();i++){
            mp[s[i]]++;
        }
        unordered_map <char,int>::iterator it = mp.begin();
        int unique = 0,length = 0;
        while(it!=mp.end()){
            if((it->second)%2){
                unique = 1;
                length+=(it->second)-1;
            }
            else{
                length+=it->second;
            }
            ++it;
        }
        return length+unique;
    }
};

File: 409-longest-palindrome/longest-palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        d = {}
        for i in s:
            if i not in d:
                d[i]=1
            else:
                d[i]+=1
        l = 0
        unique = 0
        for i in d.keys():
            if d[i]%2==0:
                l+=d[i]
            else:
                l+=d[i]-1
                unique=1
        return unique+l
        